pkg/vault/secret/engines/pki/secret.go

```diff
@@ -0,0 +1,88 @@
+package pki
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	vaultapi "github.com/hashicorp/vault/api"
+	"github.com/kubevault/operator/pkg/vault/secret"
+	"github.com/pkg/errors"
+)
+
+const UID = "PKI"
+
+type SecretInfo struct {
+	// Specifies the path where secret engine is enabled
+	Path string
+
+	// Specifies the role
+	Role string
+
+	Client *vaultapi.Client
+}
+
+func New() *SecretInfo {
+	return &SecretInfo{}
+}
+
+func NewSecretManager() secret.SecretManager {
+	return &SecretInfo{}
+}
+
+func NewSecretManagerWithOptions(c *vaultapi.Client, opts map[string]string) secret.SecretManager {
+	s := &SecretInfo{}
+	s.Client = c
+	if val, ok := opts[secret.RoleKey]; ok {
+		s.Role = val
+	}
+	if val, ok := opts[secret.PathKey]; ok {
+		s.Path = val
+	}
+	return s
+}
+
+func NewSecretGetter(vc *vaultapi.Client, path string, role string) secret.SecretGetter {
+	return &SecretInfo{
+		Client: vc,
+		Path:   path,
+		Role:   role,
+	}
+}
+
+func (s *SecretInfo) SetOptions(c *vaultapi.Client, opts map[string]string) error {
+	s.Client = c
+	if val, ok := opts[secret.RoleKey]; ok {
+		s.Role = val
+	}
+	if val, ok := opts[secret.PathKey]; ok {
+		s.Path = val
+	}
+	return nil
+}
+
+func (s *SecretInfo) GetSecret() ([]byte, error) {
```

Comment:
PKI issue should be like this https://github.com/kubevault/csi-driver/blob/master/vault/secret/engines/pki/secret.go#L59